scripts/adr-doctor.sh

```
@@ -0,0 +1,235 @@
+
+# --- no_legacy_subdir (INV-A12) ---
+note "no_legacy_subdir (INV-A12)"
+if [ -d "$ADR_DIR/legacy" ]; then
+  check_fail "$ADR_DIR/legacy must not exist (stubs MUST stay flat)"
+fi
+
+# --- stub_links_real (INV-A12) ---
+note "stub_links_real (INV-A12)"
+for stub in "$ADR_DIR"/[0-9][0-9][0-9][0-9]-*.md; do
+  [ -f "$stub" ] || continue
+  target=$(grep -oE '\[`[a-z0-9-]+\.md`\]' "$stub" | head -1 | tr -d '[]`' )
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)